core/bin/zksync_server/src/main.rs

```diff
@@ -140,18 +140,32 @@ fn main() -> anyhow::Result<()> {
         }
     };
 
-    let secrets: Secrets = match opt.secrets_path {
-        Some(path) => {
-            let yaml =
-                std::fs::read_to_string(&path).with_context(|| path.display().to_string())?;
-            decode_yaml_repr::<zksync_protobuf_config::proto::secrets::Secrets>(&yaml)
-                .context("failed decoding secrets YAML config")?
+    let secrets: Secrets = {
+        let consensus_secrets =
+            config::read_consensus_secrets().context("config::read_consensus_secrets()")?;
+
+        match opt.secrets_path {
+            Some(path) => {
+                let yaml =
+                    std::fs::read_to_string(&path).with_context(|| path.display().to_string())?;
+
+                let mut secrets =
+                    decode_yaml_repr::<zksync_protobuf_config::proto::secrets::Secrets>(&yaml)
+                        .context("failed decoding secrets YAML config")?;
+
+                // Since secrets can be defined in two files, allow the convention of only specifying consensus secrets in one place.
+                if secrets.consensus.is_none() {
```

Comment:
IMO it would be better to fail if both ops.secrets_path and the env var with consensus secrets are specified (i.e. disallow coexistence of both config formats altogether). At the very least we should fail if both secrets.consensus.is_some() and env var with consensus secrets are specified.

Reply:
> it would better to fail if both ops.secrets_path and the env var with consensus secrets are specified

As I understand they want to move to files from env vars, which suggests that the `ops.secrets_path` is not going away, which would mean this would almost surely fail.

> At the very least we should fail if both

Okay, sounds good. Maybe with a check that they are actually different secrets, so it can be backwards compatible with the current gitops.